src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.cpp

```diff
@@ -0,0 +1,86 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp"
+
+#include <array>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/LeviCivitaIterator.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace GeneralizedHarmonic::BoundaryConditions::detail {
+template <size_t VolumeDim, typename DataType>
+void set_dt_v_zero_constraint_preserving(
+    gsl::not_null<tnsr::iaa<DataType, VolumeDim, Frame::Inertial>*>
+        bc_dt_v_zero,
+    const tnsr::I<DataType, VolumeDim, Frame::Inertial>&
+        unit_interface_normal_vector,
+    const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>&
+        four_index_constraint,
+    const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_zero,
+    const std::array<DataType, 4>& char_speeds) noexcept {
+  ASSERT(get_size(get<0, 0, 0>(*bc_dt_v_zero)) ==
+             get_size(get<0>(unit_interface_normal_vector)),
+         "Size of input variables and temporary memory do not match: "
```

Comment:
Perhaps clearer would be `"Sizes of unit_interface_normal_vector (" <<  get_size(get<0, 0, 0>(*bc_dt_v_zero)) << ") and bc_dt_v_zero (" << get_size(get<0>(unit_interface_normal_vector)) << ") do not match"`.

But I wonder, if the sizes don't match, why not just do a destructive resize to make the size match?

Reply:
indeed, changed this to reallocate and not throw an error. Cant use destructive resize as the input points to these functions would typically be references to memory inside of TempBuffers which would be non-owning.